js/data.js

```diff
@@ -36,7 +37,13 @@ var metadata = {
 };
 
 if (localStorage.getItem(localStorageKey) !== null) {
-  data = JSON.parse(localStorage.getItem(localStorageKey));
+  var referenceDataKeys = Object.keys(data);
+  var tmpData = JSON.parse(localStorage.getItem(localStorageKey));
+  if (JSON.stringify(referenceDataKeys) === JSON.stringify(Object.keys(tmpData))) {
```

Comment:
Using JSON.stringify this way works, but is not efficient and only works in certain situations. It's best to do a quick comparison.

```js
function equalArrays(first, second) {
  if (first.length !== second.length) return false;
  for (var i = 0; i < first.length; i++) {
    if (first[i] !== second[i]) return false;
  }
  return true;
}
```

Reply:
Thanks, not sure why I didn't just do this from the start